ai-chat-assistant/repo_scanner.py

```diff
@@ -0,0 +1,135 @@
+import datetime
+import os
+
+from autokitteh import github, google
+from autokitteh.slack import slack_client
+from helpers import get_sheets_data
+
+
+g = github.github_client("github_conn")
+slack = slack_client("slack_conn")
+sheets = google.google_sheets_client("googlesheets_conn")
+
+
+SHEET_ID = os.getenv("SHEET_ID")
+SHEET_NAME = os.getenv("SHEET_NAME")
+
+
+def find_unanswered_comments(
+    repo_name: str, user: str
+) -> list[tuple[str, str, str, str]]:
+    """Entrypoint for finding unanswered comments."""
+    print("Finding unanswered messages...")
+    sheets_data = get_sheets_data(SHEET_NAME)
+    comment_ids_set = set()
+    if "values" in sheets_data:
+        for row in sheets_data["values"]:
+            comment_ids_set.add(int(row[0]))
+    return get_github_comments(comment_ids_set, repo_name, user)
+
+
+def get_github_comments(
+    comment_ids_set: set,
+    repo_name: str,
+    user: str,
+) -> list[tuple[str, str, str, str]]:
+    """Get all comments from GitHub that haven't been responded to in over 24 hours."""
+    repo = g.get_repo(repo_name)
+    pulls = repo.get_pulls(state="open")
+
+    user_email = get_email_by_slack_user_id(user)
+    github_user_id = get_github_user_id_by_email(user_email)
+
+    unresponded = []
+
+    # Process each pull request
+    for pr in pulls:
+        # Combine issue and inline comments
+        issue_comments = list(pr.get_issue_comments())
+        inline_comments = list(pr.get_comments())
+        for comments, is_inline in [(issue_comments, False), (inline_comments, True)]:
+            process_comments(
+                comments,
+                comment_ids_set,
+                unresponded,
+                github_user_id,
+                is_inline=is_inline,
+            )
+
+    return unresponded
+
+
+def process_comments(
+    comments: list,
+    comment_ids_set: set,
+    unresponded: list,
+    github_user_id: str,
+    is_inline: bool = False,
+):
+    """Process either issue comments (PR discussion) or inline review comments."""
+    for comment in comments:
+        if should_skip_comment(comment, github_user_id):
+            continue
+
+        if has_been_responded_to(comment, comments, is_inline, github_user_id):
+            continue
+
+        print()
+        if comment.id not in comment_ids_set:
+            unresponded.append(
+                (comment.id, comment.user.login, comment.body, comment.html_url)
+            )
+            comment_ids_set.add(comment.id)
+
+
+def has_been_responded_to(
+    target_comment, potential_responses, is_inline, github_user_id
+):
+    # Skip if comment is less than 24 hours old
```

Comment:
Also add in other comments in this function.
```suggestion
    # Skip if comment is less than 24 hours old.
```

Reply:
Done